### ISSUE TITLE
building with `--no-verify-sdk-hash` requires a (unused) git internal file to exist

### ISSUE THREAD

**Issue by selfisekai** [dart-lang/sdk#52295]
when `--no-verify-sdk-hash` is used with `//tools/gn.py`, gn does not determine the last commit timestamp with git. a file in the path `.git/logs/HEAD` still has to exist though, because it is considered as needed for the build by ninja. there's no build instructions for it either, making ninja go "Rebuilding" like 100 times in a loop and then failing.

**Comment by sortie** [comment]
@selfisekai Do you have a solution in mind that you might want to contribute? Sounds like the gn code shouldn't emit that dependency when --no-verify-sdk-hash

**Comment by selfisekai** [comment]
no, I currently workaround that with mkdir and touch on that path

**Comment by selfisekai** [comment]
ok, the biggest problem I see is here: https://github.com/dart-lang/sdk/blob/214a98e1dd29c2a3aeef2b9b2d6bb2f053df24bf/third_party/binaryen/BUILD.gn#L62
there needs to be a way to make some other kind of stamp. put it in gclient gn args and make `//build/config/gclient_args.gni` a stamp file?
